### ISSUE TITLE
GitHub Actions - Update workflows using Node JS 16 actions to support Node JS 20

### ISSUE THREAD

**Issue by achyuthajoy** [Automattic/woocommerce-payments#8129]
### Describe the bug

On our workflows we're seeing notices from GitHub indicating NodeJS 16 actions are deprecated, prompting to upgrade action versions to support NodeJS 20. More info can be found here - For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-node@v3, actions/cache@v3. 
```

Ref: https://github.com/Automattic/woocommerce-payments/actions/runs/7775675801

**Comment by zmaglica** [comment]
@deepakpathania I will move this issue to the team Harmony. I am aware that team Harmony might not want to do the dependency upgrade for every package we use, so feel free to ping other teams or me, if you think that someone else might be a better fit for this issue. I will move it to the Other Teams queue as a part of our triage process.

**Comment by danielmx-dev** [comment]
@deepakpathania I wanted to check-in about the status of this issue, since it's currently blocking  #8581. Is this something that will be picked up soon by your team? Otherwise, our team (Heisenberg) can start working on it so we can unblock our other issue.

**Comment by achyuthajoy** [comment]
@danielmx-dev This issue is about updating the actions provided by GitHub such as `actions/checkout`, `actions/cache` etc to Node 20. This has nothing to do with the node version of the plugin.

This issue is not prioritized yet. Feel free to pick it up, but this is not a blocker for https://github.com/Automattic/woocommerce-payments/issues/8581

**Comment by deepakpathania** [comment]
> Is this something that will be picked up soon by your team?

This focus area is now owned by Architects so looping in @jrodger for assisting with prioritisation. Thanks.

**Comment by danielmx-dev** [comment]
I created a new issue to update the node version of the plugin itself https://github.com/Automattic/woocommerce-payments/issues/9116
